### PR TITLE
Update libbpf to v1.0.0

### PR DESCRIPTION
**Pull request by saschagrunert**
Updates libbpf to the latest release: https://github.com/libbpf/libbpf/releases/tag/v1.0.0

### CODE REVIEW

**Comment by grantseltzer** [comment]
Thanks @saschagrunert ! I'm really excited for this, hoping to have time to review and merge this by EOD tomorrow and a new release sometime next week once we're sure we've removed everything that's deprecated and firmly standardized error checking the way 1.0 does it. 

I'll also share publicly our roadmap towards libbpfgo 1.0 which will certainly have to include a full implementation of the libbpf API. I gotta setup a public office hours so we (all contributors/maintainers) can discuss that and other exciting things we can do such as being part of the libbpf project more officially. Will keep you posted!

**Comment by kakkoyun** [comment]
> I'll also share publicly our roadmap towards libbpfgo 1.0 which will certainly have to include a full implementation of the libbpf API.

As Parca Agent maintainers, we would be more than happy to give a hand on this effort. 

> I gotta setup a public office hours so we (all contributors/maintainers) can discuss that and other exciting things we can do such as being part of the libbpf project more officially.

❤️ It will be amazing to have office hours!

**Comment by grantseltzer** [comment]
I'm presently going through libbpfgo to make sure we're not using any now removed APIs and are handling errors correctly. There's also a bunch of new APIs and things that we can do differently in 1.0 that'll i'm going to be creating issues for but won't hold up this PR (but will be before the next libbpfgo release).

**Comment by grantseltzer** [comment]
> > I'll also share publicly our roadmap towards libbpfgo 1.0 which will certainly have to include a full implementation of the libbpf API.
> 
> As Parca Agent maintainers, we would be more than happy to give a hand on this effort.

:heart: 

> 
> > I gotta setup a public office hours so we (all contributors/maintainers) can discuss that and other exciting things we can do such as being part of the libbpf project more officially.
> 
> heart It will be amazing to have office hours!

Will let you know very soon!

**Comment by grantseltzer** [comment]
Alright I've opened https://github.com/aquasecurity/libbpfgo/pull/225 which we'll merge at the same time as this one. For some reason github doesn't let me tag people out of the aqua org for review but @kakkoyun @javierhonduco @saschagrunert your feedback is always greatly appreciated!

File: libbpfgo.go
Comment (grantseltzer) [inline libbpfgo.go:1034]: Why is this?

File: selftest/cgroup/main.bpf.c
Comment (grantseltzer) [inline selftest/cgroup/main.bpf.c:17]: Where can I find a reference to accepted SEC macro handles?

File: selftest/iterators/main.go
Comment (grantseltzer) [inline selftest/iterators/main.go:54]: I don't think we need this

File: libbpfgo.go
Comment (saschagrunert) [reply libbpfgo.go:1034]: That's the return code for the function call, one of the map tests failed because of this change.

File: selftest/iterators/main.go
Comment (saschagrunert) [reply selftest/iterators/main.go]: Removed as suggested :+1:

File: selftest/cgroup/main.bpf.c
Comment (saschagrunert) [reply selftest/cgroup/main.bpf.c:17]: There: https://github.com/libbpf/libbpf/blob/a0d1e22c7790f22809d33b3c5c1e3ded89157cc8/src/libbpf.c#L8472-L8554

File: libbpfgo.go
Comment (grantseltzer) [reply libbpfgo.go:1050]: Got it! I see now. i guess it's redundant to check `errno == C.ENOENT` and `errC = -2`. Do you think we should simplify to just the following?: 
```suggestion
	if errC == -2 {
```

Comment (grantseltzer) [reply libbpfgo.go:1050]: This is actually outside the scope of this PR, so I'd say leave it for now and we can reassess before the next release

**Review by grantseltzer — APPROVED** [review]
Let's do this!

**Comment by grantseltzer** [comment]
Thanks again @saschagrunert, Ill be looking to cut a release by middle of next week.

**Pull request merged**
